docs/storage/commit_log/main.go

```diff
@@ -68,28 +70,51 @@ func increment(s string) string {
 	return string(append([]byte(increment(prefix)), 'A'))
 }
 
+type atomicBool struct {
```

Comment:
BTW, there's `sync/atomic` which provides atomic ints, you could implement this in terms of those with `CompareAndSwap()` and save a few lines perhaps?

Reply:
Atomics are best avoided; renamed to `lockedBool` to reduce the temptation in future.